pageserver/src/tenant/size.rs

```diff
@@ -37,13 +62,19 @@ struct TimelineInputs {
     horizon_cutoff: Lsn,
     #[serde_as(as = "serde_with::DisplayFromStr")]
     pitr_cutoff: Lsn,
+
+    /// Cutoff point based on GC settings
     #[serde_as(as = "serde_with::DisplayFromStr")]
     next_gc_cutoff: Lsn,
+
+    /// Cutoff point calculated from the user-supplied 'max_retention_period'
+    #[serde_as(as = "Option<serde_with::DisplayFromStr>")]
+    retention_param_cutoff: Option<Lsn>,
 }
 
 /// Gathers the inputs for the tenant sizing model.
 ///
-/// Tenant size does not consider the latest state, but only the state until next_gc_cutoff, which
+/// Tenant size does not consider the latest state (???), but only the state until next_gc_cutoff, which
```

Comment:
Before #2781 I don't see how it would had reflected the latest state, by which I meant logical size at `last_record_lsn`, which I had in the gathering originally and in discussions it was expected to be there (esp. in relation to early `last_record_lsn < gc_cutoff` tenants).